### PR TITLE
feat(embed): embed lookup route

### PR DESCRIPTION
**Pull request by joshuaeilers**
This PR sets up the datahub-web side for the Tableau chrome extension. The datahub web side is quite simple. You pass it an encoded url, and it will run a query against our possible external url fields for a match. And if a match is found, it renders the chrome extension page. I added a not found analytics event as well for tracking failures / unexpected states a user may find themselves in.

Screenshot 2023-05-18 at 1 06 12 PM

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/src/app/embed/EmbedLookup.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    useEffect(() => {
+        if (destinationUrl) {
+            console.log('routing to', destinationUrl, 'from', history.location);
+        }
+    }, [destinationUrl, history.location]);
+
+    if (error) return <div>{JSON.stringify(error)}</div>;
+    if (notFound) return <div>Not found</div>;
+    if (multipleResults) return <div>Multiple results</div>;
+    if (loading || !destinationUrl) return <div>Loading</div>;
```
Comment (joshuaeilers) [inline datahub-web-react/src/app/embed/EmbedLookup.tsx:59]: Show an error if data finished loading but we had no url. ie. mayabe create an error in the memo if unable to construct a valid url from the entity. ie. More validation on the results array or missing entity etc

File: datahub-web-react/src/app/analytics/event.ts [hunk 2]
```diff
@@ -519,6 +520,12 @@ export interface EmbedProfileViewInDataHubEvent extends BaseEvent {
     entityUrn: string;
 }
 
+export interface EmbedLookupNotFoundEvent extends BaseEvent {
+    type: EventType.EmbedLookupNotFoundEvent;
+    url: string;
+    reason: 'NO_ENTITY_FOUND' | 'MULTIPLE_ENTITIES_FOUND';
```
Comment (joshuaeilers) [inline datahub-web-react/src/app/analytics/event.ts:526]: Totally up for debate 😆

Comment (chriscollins3456) [reply datahub-web-react/src/app/analytics/event.ts:526]: i think these are good! nit: can we make these strings constants to reuse?

**Review by chriscollins3456 — APPROVED** [review]
nice man this looks great!

File: datahub-web-react/src/app/analytics/event.ts (hunk 2 above)
Comment (joshuaeilers) [reply datahub-web-react/src/app/analytics/event.ts:526]: I think I'd prefer a type we can import vs a constant. That way we get compiler support, and we can still do refactoring shortcuts in vscode.

**Pull request merged**